standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1053,7 +1054,16 @@ public void abortTxn(AbortTxnRequest rqst) throws NoSuchTxnException, MetaExcept
           }
           raiseTxnUnexpectedState(status, txnid);
         }
-        abortTxns(dbConn, Collections.singletonList(txnid), true, isReplayedReplTxn);
+
+        if (isReplayedReplTxn) {
+          errorMsg = TxnErrorMsg.ABORT_REPLAYED_REPL_TXN;
+        } else if (isHiveReplTxn) {
+          errorMsg = TxnErrorMsg.ABORT_DEFAULT_REPL_TXN;
+        } else if (rqst.isSetErrorCode() && rqst.getErrorCode() == TxnErrorMsg.ABORT_ROLLBACK.getErrorCode()) {
```

Comment:
what is this? why not 
````
if (rqst.isSetErrorCode() {
  errorMsg = TxnErrorMsg.findByValue(rqst.getErrorCode())
}
````

Reply:
should isReplayedReplTxn override rqst.getErrorCode()?

Reply:
Updated the condition to - 
`if (rqst.isSetErrorCode() {
   errorMsg = TxnErrorMsg.getErrorMsg(rqst.getErrorCode())
}`

As far as isReplayedReplTxn is concerned. There was no such intention of keeping it first in if else block. 
Please note that isReplayedReplTxn & isHiveReplTxn is set only when request is passed through replRollbackTxn() call in the MetastoreClient.